File: apps/api/AssetManagement.Api/Models/SavedView.cs
namespace AssetManagement.Api.Models;

public class SavedView
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public required string EntityType { get; set; }
    public required string Name { get; set; }
    public bool IsDefault { get; set; }
    public required string Configuration { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public User User { get; set; } = null!;
}

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/model/SavedView.kt
package com.assetmanagement.api.model

import jakarta.persistence.*
import java.time.Instant
import java.util.*

@Entity
@Table(
    name = "saved_views",
    indexes = [Index(columnList = "user_id, entity_type")]
)
class SavedView(
    @Id
    @Column(name = "id", columnDefinition = "CHAR(36)")
    var id: UUID = UUID.randomUUID(),

    @Column(name = "user_id", nullable = false, columnDefinition = "CHAR(36)")
    var userId: UUID = UUID.randomUUID(),

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    var user: User? = null,

    @Column(name = "entity_type", nullable = false)
    var entityType: String = "",

    @Column(name = "name", nullable = false)
    var name: String = "",

    @Column(name = "is_default", nullable = false)
    var isDefault: Boolean = false,

    @Column(name = "configuration", nullable = false, columnDefinition = "TEXT")
    var configuration: String = "",

    @Column(name = "created_at", nullable = false)
    var createdAt: Instant = Instant.now(),

    @Column(name = "updated_at", nullable = false)
    var updatedAt: Instant = Instant.now()
)
